monitoring/etcd/etcd.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/gravitational/satellite/monitoring"
+	"github.com/gravitational/trace"
+	"github.com/prometheus/client_golang/prometheus"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	namespace             = "etcd"
+	collectMetricsTimeout = 5 * time.Second
+)
+
+var (
+	followersLatency = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "etcd_followers_latency"),
+		"Bucketed histogram of latency time (s) between ETCD leader and follower",
+		//			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 13), // buckets from 0.0001 till 4.096 sec
+		[]string{"followerName"}, nil,
+	)
+
+	followersRaftFail = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "etcd_followers_raft_fail"),
+		"Counter of Raft RPC failed requests between ETCD leader and follower",
+		[]string{"followerName"}, nil,
+	)
+
+	followersRaftSuccess = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "etcd_followers_raft_success"),
+		"Counter of Raft RPC successful requests between ETCD leader and follower",
+		[]string{"followerName"}, nil,
+	)
+)
+
+// LeaderStats is used by the leader in an etcd cluster, and encapsulates
+// statistics about communication with its followers
+// reference documentation https://github.com/coreos/etcd/blob/master/etcdserver/stats/leader.go
+type LeaderStats struct {
+	// Leader is the ID of the leader in the etcd cluster.
+	Leader    string                    `json:"leader"`
+	Followers map[string]*FollowerStats `json:"followers"`
```

Comment:
do you need pointers? Otherwise, I'd go with values.
Also, missing comments on exported fields